### DISCUSSION TITLE
Get configuration root in v5

### DISCUSSION THREAD

**Discussion by jamesholcomb** [feathersjs/feathers#3051] [Q&A] [answered]
How do I get the root configuration object (and destructure from it)?

In v4, you could do:

```js
import configuration from "@feathersjs/configuration"

const config = configuration()()
const { db } = config
```

...but this no longer works.  I understand that you can call `app.get(key)` but trying to minimize refactoring for the migration.

**Comment by daffl** [comment, marked as answer]
I don't think that usage has changed, it should still return the configuration when called that way (see https://github.com/feathersjs/feathers/blob/dove/packages/configuration/src/index.ts#L13). What is it returning now?

**Comment by jamesholcomb** [comment]
After more digging, it seems related to `dotenv` initialization.  The `config` var from above does indeed have the json values from the file, but env var substitution did not occur.

I utilize `ts-node` and `swc`.  App start is:

```sh
NODE_OPTIONS='-r dotenv/config --no-warnings' ts-node src/
```

...so `dotenv` should be setup.   I'm walking through MIGRATING.md on a v4 project.

**Reply by daffl** [reply]
In that case https://feathersjs.com/guides/migrating.html#configuration might be important. You'll likely have to create a `custom-environment-variables.json` file for the environment variables you want to use, this is no longer done automatically.

**Reply by jamesholcomb** [reply]
👍 that was the culprit thanks.  I recall initially scanning through this section and reading "to instead rely on the functionality already provided and battle tested by the underlying node-config" but stopped there thinking no further action would be required 🤷

**Comment by axmad22** [comment]
You can try switching the config files from json to cjs format then you can have access to **env** variables in the config like so

```js
module.exports = {
    host: process.env.HOST || '127.0.0.1',
    port: process.env.PORT || 3000,
  }
```
